_includes/seo.html

```diff
@@ -4,13 +4,20 @@
 {%- endif -%}
 {%- assign seo_url = seo_url | default: site.github.url -%}
 
-{% assign title_separator = site.title_separator | default: '-' | replace: '|', '&#124;' %}
 
-{%- if page.title -%}
-  {%- assign seo_title = page.title | append: " " | append: title_separator | append: " " | append: site.title -%}
+{%- unless site.title_separator == false -%}
+  {%- assign title_separator = site.title_separator | default: '-' | replace: '|', '&#124;' -%}
+  {%- assign append_title = " " | append: title_separator | append: " " | append: site.title -%}
+{%- endunless -%}
+
+{%- if page.seo_title -%}
```

Comment:
Let's keep the current behavior intact (falling back to `site.title`) and simplify all this thing:

```diff
-{%- if page.seo_title -%}
-  {%- assign seo_title = page.seo_title -%}
-{%- elsif page.title -%}
-  {%- assign seo_title = page.title -%}
-{%- endif -%}
+{%- assign seo_title = page.seo_title | default: page.title | default: site.title -%}
```

Reply:
@iBug Did you mean like this?